src/codons.py

```diff
@@ -35,8 +35,15 @@ def split_codons(dna: str) -> list[str] | None:
     True
 
     """
-    # FIXME: Implement the function
-    return []
+    codons = []
+    dna_length = len(dna)
```

Comment:
You don't need to save the size in a variable. It is practically just as efficient to ask for `len(dna)` every time you need the length.